### ISSUE TITLE
Termination code 11

### ISSUE THREAD

**Issue by svigerske** [coin-or/Ipopt#98]
Issue created by migration from Trac.

Original creator: jakesson

Original creation time: 2009-05-25 12:59:15

Assignee: ipopt-team

Version: 

The issue is described in the mailing list post

http://list.coin-or.org/pipermail/ipopt/2009-May/001517.html

Termination code 11 results if IPOPT (when invoked from AMPL) is compiled after running the configure script without arguments. The config.log file and the resulting .nl file are attached.

The following three configure argument configurations has been tried (in addition to no arguments):

 - CXXFLAGS='-O3 -g -DNDEBUG' CFLAGS='-O3 -g -DNDEBUG' FFAGS='-O3 -g'
 - CXXFLAGS='-O -g -DNDEBUG' CFLAGS='-O -g -DNDEBUG' FFAGS='-O -g'
 - --enable-debug

In all these cases the optimization runs fine.

The tests are run with IPOPT 3.6.1 (which cannot be selected in the list below).

(changed by @andrea5w at 2009-05-26 15:18:44)

**Comment by svigerske** [comment]
Attachment [config.log](https://gist.githubusercontent.com/svigerske/e66f165660027ccb65ebff89ee31d0e9/raw/8ff3186ec11d4c13716312960e3c6b40fc804999/Ipopt%20issue%2098%20attachment%20config.log) by jakesson created at 2009-05-25 13:01:48

Log file generated by configure

**Comment by svigerske** [comment]
Attachment [bla.nl](https://gist.githubusercontent.com/svigerske/7cda0512d485bdd8c18b812f65d13762/raw/b797d3e8ab8d7881ce4734db22d49038d17b5df6/Ipopt%20issue%2098%20attachment%20bla.nl) by jakesson created at 2009-05-25 13:02:34

.nl file generated by AMPL

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2009-05-26 15:18:44

Hi Johan,

Thanks for creating the ticket.

I had a look at the config.log file, and it seems that you didn't specify the flags as you describe in this ticket, you only ran it as "./configure" and therefore Ipopt used the default compiler flags:

CXXFLAGS='-O3 -fomit-frame-pointer -pipe -DNDEBUG -pedantic-errors -Wimplicit -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas  '

I ran your bla.nl example with 3.6.1, and I had no issues, it ran well, and valgrind did not detect any errors.  I used the MA27 linear solver (from your output it is not clear which one you used - maybe you didn't include one and this might be the issues?  Did "make test" work?)

You are using the GCC 4.3.2 compilers.  I have not tried to compile Ipopt with those, and it could be that this is the issue.  I have successfully used Ipopt with GCC 4.1.2 and 4.2.1.  As we discussed in the mailing list, the segfault happens in a function of the ASL, which is not Ipopt code, and you might have to talk to the developer of that software...

Not sure what I can add at this point...

Regards,

Andreas

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2009-05-26 15:18:44

Changing assignee from ipopt-team to @andrea5w.

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2009-05-26 15:18:44

Changing status from new to assigned.

**Comment by svigerske** [comment]
Comment by jakesson created at 2009-05-28 17:40:14

Andreas,

Just to clarify: In the three cases of different configure arguments listed in the description I had no problems. However, I got termination code 11 when I compiled after invoking configure without any arguments. The config.log that I attached was the one generated in this latter cases, so it is correct that no arguments were given to configure. Since the other three cases did not render any problem (no termination code 11), I did not attach the corresponding configure.log files.

Best
/Johan
